debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java

```diff
@@ -120,7 +120,7 @@ else if (connectorConfig.getSnapshotIsolationMode() == SnapshotIsolationMode.EXC
 
                     LOGGER.info("Locking table {}", tableId);
 
-                    String query = String.format("SELECT * FROM [%s].[%s] WITH (TABLOCKX)", tableId.schema(), tableId.table());
+                    String query = String.format("SELECT TOP(0) * FROM [%s].[%s] WITH (TABLOCKX)", tableId.schema(), tableId.table());
```

Comment:
So will this actually make sure the table is locked (was surprised that it isn't `TOP(1)`)?

Reply:
@gunnarmorling Yes, I've tried it with SQL client and it really locks the whole table. Unless the tx is committed the read access is blocked form other TX.

Reply:
Ok, cool. Nice improvement!